coins/ethereum/contracts/Router.sol

```diff
@@ -0,0 +1,101 @@
+//SPDX-License-Identifier: AGPLv3
+pragma solidity ^0.8.0;
+
+import "./Schnorr.sol";
+
+contract Router is Schnorr {
+    // contract owner
+    address owner;
```

Comment:
... do we want to replace this with just a PublicKey? I understand why the address form exists. It's just a variable we immediately kill off yet have to deal with FUD for (it has an owner!!!). I'd kinda rather overload a PublicKey variable (magic parity which will be considered invalid, yet signals px is an address for setPublicKey?). This also would save a minute amount of gas.

Reply:
right, so do you mean replace this type with PublicKey or just remove it entirely and set the `PublicKey publicKey` in the constructor? I tried setting it in the constructor, but due to limitations with ethers-rs I was unable to actually call the constructor and pass in the keys :// might be able to now, since I believe they made some updates, can try again